### ISSUE TITLE
Spotify API changes break Spotty plugin

### ISSUE THREAD

**Issue by rtfmjoey** [michaelherger/Spotty-Plugin#152]
Hi, 

It seems that the newest API changes break the Spotty plugin.

I'm getting the dreaded 'Failed to get access token' and looking at the logs I notice the following:

`[24-11-27 18:55:01.2391] Plugins::Spotty::API::__ANON__ (1361) Trying to read from cache for browse/featured-playlists?country=NL&limit=50&locale=en_US&timestamp=2024-11-27T18%3A50%3A00
[24-11-27 18:55:01.2392] Plugins::Spotty::API::__ANON__ (1370) API call: browse/featured-playlists?country=NL&limit=50&locale=en_US&timestamp=2024-11-27T18%3A50%3A00
[24-11-27 18:55:01.2939] Plugins::Spotty::API::__ANON__ (1467) error: 404 Not Found
[24-11-27 18:55:01.2963] Plugins::Spotty::OPML::__ANON__ (189) Failed to get featured playlists and/or token - do not continue

`

get-featured-playlists is removed from the API, see https://developer.spotify.com/blog/2024-11-27-changes-to-the-web-api for details.

**Comment by cpedrero** [comment]
Same problem 😵‍💫

**Comment by michaelherger** [comment]
Oh wow... and they don't even inform their developers ahead of time?!? That's really bad. I'll see what I can do.

Could you please confirm what Spotty versions you're using?

**Comment by rtfmjoey** [comment]
Logitech Media Server Version: 8.5.2
Spotty version: 4.11.3

I've tried with the new test spotty but ran into the same issue so I've reverted and am using Spotify Connect for the time being.
Als created a new app in Spotify dev, added users etc and configured the new app in Spotty but this also resulted in the same issue.

**Comment by michaelherger** [comment]
Well, as the links blog article suggests they simply remove access to some crucial endpoints without notice. One _possible_ workaround is to remove your own client ID. But that will cause other issues in the long run...

**Comment by oli101010** [comment]
Thanks for the workaround! Removing the client ID resolved the issue—Spotty can now browse and play again as expected.

**Comment by fortender** [comment]
We ran into the same issues. The problem arises from `OPML.handleFeed` relying on a successful call to the `/browse/featured-playlists` endpoint, as you guys already stated above.

A quick'n'dirty solution was to remove the following lines https://github.com/michaelherger/Spotty-Plugin/blob/8b10ec3804dbc17833a54b1433cf5fe8ab3bff15/OPML.pm#L190-L197

You find the file in `%ProgramData%\Squeezebox\Cache\InstalledPlugins\Plugins\Spotty\OPML.pm` (Windows). The crucial part is the return here. This stops `handleFeed` from continuing to build the menu structure and prints the error. It seems like it's also used as an indicator of whether or not our token is valid, but we really should handle that in another way.

I also provided a PR #153 that removes the entire call and the function itself.
